allennlp/tests/models/encoder_decoders/simple_seq2seq_test.py

```diff
@@ -77,28 +77,3 @@ def test_decode_runs_correctly(self):
         output_dict = self.model(**training_tensors)
         decode_output_dict = self.model.decode(output_dict)
         assert "predicted_tokens" in decode_output_dict
-
-    def test_greedy_decode_matches_beam_search(self):
```

Comment:
I actually quite like having this test to ensure that the beam search and greedy search are in sync. Did you remove it so that `_forward_loop` doesn't need to make predictions? Honestly, I'd say it's worth leaving in the predictions just for this test. I'm open to being convinced otherwise, but I'd need to see some benchmarks showing a large performance hit.

Reply:
That's a good point. I had initially removed it because I thought it would be confusing to have `_forward_loop` create predictions, only to be overridden by the predictions from beam search. I put that back in though and just added a note that we really only use those predictions for testing